llvm/lib/Target/SBF/SBFInstrInfo.td

```diff
@@ -302,6 +317,22 @@ let Constraints = "$dst = $src2" in {
         defm SRL : ALU<SBF_RSH, "rsh", srl>;
         defm XOR : ALU<SBF_XOR, "xor", xor>;
         defm SRA : ALU<SBF_ARSH, "arsh", sra>;
+
+        let Predicates = [SBFv2] in {
+            def HOR :  ALU_RI<SBF_ALU64, SBF_HOR,
+                        (outs GPR:$dst),
+                        (ins GPR:$src2, i32imm:$imm),
+                        "hor64 $dst, $imm",
+                        []>;
+            let DecoderNamespace = "AddrLoad" in {
+              def HOR_addr :  ALU_RI<SBF_ALU64, SBF_HOR,
+                                      (outs GPR:$dst),
+                                      (ins GPR:$src2, u64imm:$imm),
+                                      "hor64 $dst, $imm",
+                                      []>;
```

Comment:
While this isn't incorrect per-se, we do not need two different instructions defined for each case (different selection patterns are enough). Optional change.

Reply:
I couldn't see how to do it with patterns. The two instructions have different operand types, and I get and error if I use only `HOR`.